src/altimeter.rs

```diff
@@ -0,0 +1,273 @@
+use bmp388::{config::{FifoConfig, OversamplingConfig}, Blocking, PowerControl, BMP388};
+use heapless::Vec;
+use core::{cell::RefCell, ptr::addr_of_mut, sync::atomic::AtomicBool};
+use cortex_m::interrupt::Mutex;
+use stm32f4xx_hal::{
+    i2c::dma::{
+            I2CMasterHandleIT, I2CMasterWriteReadDMA, I2cCompleteCallback
+        }, interrupt
+};
+use stm32f4xx_hal::i2c;
+use stm32f4xx_hal::timer::delay::SysDelay;
+use crate::futures::YieldFuture;
+use crate::pins::Altimeter;
+use crate::bmp581::I2c1Handle;
+use crate::bmp581::BMP581;
+
+const BMP388_FRAME_COUNT: usize = 73;
+const BMP388_BUF_SIZE:    usize = 512;
+const BMP581_FRAME_COUNT: usize = 16;
+const BMP581_BUF_SIZE:    usize = BMP581_FRAME_COUNT * 6;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Default)]
+pub struct PressureTemp {
+    pub pressure: u32,
+    pub temperature: u32,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct FifoFrames([PressureTemp; ALTIMETER_FRAME_COUNT]);
+
+impl Default for FifoFrames {
+    fn default() -> Self {
+        FifoFrames([PressureTemp::default(); BMP388_FRAME_COUNT])
+    }
+}
+
+impl From<[PressureTemp; ALTIMETER_FRAME_COUNT]> for FifoFrames {
+    fn from(frames: [PressureTemp; ALTIMETER_FRAME_COUNT]) -> Self {
+        FifoFrames(frames)
+    }
+}
+
+impl Into<[PressureTemp; ALTIMETER_FRAME_COUNT]> for FifoFrames {
+    fn into(self) -> [PressureTemp; ALTIMETER_FRAME_COUNT] {
+        self.0
+    }
+}
+
+impl<'a> Into<&'a [PressureTemp; ALTIMETER_FRAME_COUNT]> for &'a FifoFrames {
+    fn into(self) -> &'a [PressureTemp; ALTIMETER_FRAME_COUNT] {
+        &self.0
+    }
+}
+
+pub struct BMP388Wrapper {
+    pub bmp: BMP388<I2c1Handle, Blocking>
+}
```

Comment:
Could you derive Deref & DerefMut instead of making it pub?

Reply:
Solved by moving the impl of `AltimeterFifoDMA` for `BMP581` to `bmp581.rs`